### PR TITLE
Basic credentials for Google Cloud Storage

### PR DESCRIPTION
**Pull request by coryan**
Some classes to support service account credentials.

### CODE REVIEW

File: storage/client/credentials.cc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      "/.config/gcloud/application_default_credentials.json";
+#endif
+  return suffix;
+}
+}  // anonymous namespace
+
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+std::shared_ptr<Credentials> GoogleDefaultCredentials() {
+  // There are probably more efficient ways to do this, but meh, the strings
```
Comment (houglum) [inline storage/client/credentials.cc]: Similar to the order checked in other client libraries (looking at "GetApplicationDefaultAsync()" in [1]), can we check for the GOOGLE_APPLICATION_CREDENTIALS env var first, and then fall back to the gcloud ADC file?

Also, `root` ends up being `NULL` if "HOME" is not a defined env var for some weird reason (e.g. see [2]).

[1] https://developers.google.com/api-client-library/dotnet/reference/1.9.3/classGoogle_1_1Apis_1_1Auth_1_1OAuth2_1_1GoogleCredential
[2] https://github.com/angular/angular-cli/issues/6806

Comment (coryan) [reply storage/client/credentials.cc]: On `GOOGLE_APPLICATION_CREDENTIALS`:  yes, but that is what you are going to design and implement, I am just implementing the easy part to get going :grin:  Created #522, but that is one of many other things to do, for example, also need to use GCE metadata if available.

On `HOME` not being defined: great point, thanks created #523, we can fix in this PR on a future one, up to you.

Comment (houglum) [reply storage/client/credentials.cc]: Ah, gotcha :D If this is only intended as a placeholder to provide actual credentials functionality (and only operates against gcloud's ADC file), could you leave a comment/warning to that effect at the top of the class?

**Comment by coryan** [comment]
@houglum PTAL.

**Pull request merged**